configure.ac

```diff
@@ -1071,11 +1095,18 @@ AS_CASE(["$arch,$system"],
       [default_as="${toolpref}as -arch x86_64"
       default_aspp="${toolpref}gcc -arch x86_64 -c"])],
   [amd64,solaris],
-    [default_as="${toolpref}as --64"
-    default_aspp="${toolpref}gcc -m64 -c"],
-  [i386,solaris],
-    [default_as="${toolpref}as"
-    default_aspp="${toolpref}gcc -c"],
+    [AS_CASE([$ocaml_cv_cc_vendor],
+      [sunc-*],
+        [AS_IF([test x"$SYSTEM_AS" = "x"],
+          [AC_MSG_ERROR([GNU as assembler is required.])],
+          [default_as="${toolpref}as --64"
+          default_aspp="${toolpref}cc -m64 -c"])],
+      [gcc-*],
+        [AS_IF([test x"$SYSTEM_AS" = "x"],
+          [default_as="${toolpref}gcc -m64 -c"
+          default_aspp="${toolpref}gcc -m64 -c"],
+          [default_as="${toolpref}as --64"
+          default_aspp="${toolpref}gcc -m64 -c"])])],
```

Comment:
This is probably OK.  I wonder whether we should use `$CC -c` (or something similar) as the default value for `as` and `aspp`.  This would work most of the time.  Going through the C compiler driver to invoke the assembler adds a bit of run-time but simplifies configuration quite a bit.